contracts/forecall-service/AxelarForecallService.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.9;
+import { IAxelarForecallable } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarForecallable.sol';
+import { Upgradable } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/upgradables/Upgradable.sol';
+import { IAxelarGateway } from '../interfaces/IAxelarGateway.sol';
+import { IAxelarForecallService } from '../interfaces/IAxelarForecallService.sol';
+import { IERC20 } from '../interfaces/IERC20.sol';
+
+// This should be owned by the microservice that is paying for gas.
+contract AxelarForecallService is Upgradable, IAxelarForecallService {
+    IAxelarGateway public immutable gateway;
+    address public immutable forecallOperator;
+
+    constructor(address gateway_, address forecallOperator_) {
+        gateway = IAxelarGateway(gateway_);
+        forecallOperator = forecallOperator_;
+    }
+
+    modifier onlyOperator() {
+        if (msg.sender != forecallOperator) revert NotOperator();
+
+        _;
+    }
+
+    function call(
+        bytes32 commandId,
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        address contractAddress,
+        bytes calldata payload
+    ) external onlyOperator {
+        if (contractAddress == address(0)) revert InvalidContractAddress();
+
+        if (commandId == bytes32(0)) {
+            IAxelarForecallable(contractAddress).forecall(sourceChain, sourceAddress, payload);
+        } else {
+            if (gateway.isCommandExecuted(commandId)) {
+                bytes32 payloadHash = keccak256(payload);
+
+                if (gateway.isContractCallApproved(commandId, sourceChain, sourceAddress, contractAddress, payloadHash)) {
+                    IAxelarForecallable(contractAddress).execute(commandId, sourceChain, sourceAddress, payload);
```

Comment:
not sure if we should execute. this will conflict a bit with the executor service if tries to execute at the same time (without knowledge of the forecaller)

Reply:
Executor service will not run as gas for paid for the Express Execute service